modules/calib3d/src/usac/utils.cpp

```diff
@@ -0,0 +1,187 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "../usac.hpp"
+
+namespace cv { namespace usac {
+// Performs Fisher-Yates shuffle
+void Utils::random_shuffle (RNG &rng, std::vector<int>& array) {
+    const int array_size = static_cast<int>(array.size());
+    int temp_size = static_cast<int>(array.size());
+    for (int i = 0; i < array_size; i++) {
+        const int rand_idx = rng.uniform(0, temp_size); // get random index of array of temp size
+        // decrease temp size and swap values of random index to the end (temp size)
+        std::swap(array[rand_idx], array[--temp_size]);
+    }
+}
+
+////////////////////////////////////////////////////////////////////////
+bool Math::haveCollinearPoints(const Mat &points_, const std::vector<int>& sample,
+                                      double threshold) {
+    const auto * const points = (float *) points_.data;
+    // Checks if no more than 2 points are on the same line
+    // If area of triangle constructed with 3 points is less then threshold then points are collinear:
+    //           |x1 y1 1|             |x1      y1      1|
+    // (1/2) det |x2 y2 1| = (1/2) det |x2-x1   y2-y1   0| = (1/2) det |x2-x1   y2-y1| < threshold
+    //           |x3 y3 1|             |x3-x1   y3-y1   0|             |x3-x1   y3-y1|
+    double x1, y1, x2, y2, x3, y3, X1, Y1, X2, Y2, X3, Y3;
+    int pt_idx, sample_size = static_cast<int>(sample.size());
+    for (int i1 = 0; i1 < sample_size-2; i1++) {
+        pt_idx = 4*sample[i1];
+        x1 = points[pt_idx  ]; y1 = points[pt_idx+1];
+        X1 = points[pt_idx+2]; Y1 = points[pt_idx+3];
+
+        for (int i2 = i1+1; i2 < sample_size-1; i2++){
+            pt_idx = 4*sample[i2];
+            x2 = points[pt_idx  ]; y2 = points[pt_idx+1];
+            X2 = points[pt_idx+2]; Y2 = points[pt_idx+3];
+
+            for (int i3 = i2+1; i3 < sample_size; i3++) {
+                pt_idx = 4*sample[i3];
+                x3 = points[pt_idx  ]; y3 = points[pt_idx+1];
+                X3 = points[pt_idx+2]; Y3 = points[pt_idx+3];
+
+                if (fabs((x2 - x1) * (y3 - y1) - (y2 - y1) * (x3 - x1)) / 2 < threshold)
+                    return true;
+                if (fabs((X2 - X1) * (Y3 - Y1) - (Y2 - Y1) * (X3 - X1)) / 2 < threshold)
+                    return true;
+            }
+        }
+    }
+    return false;
+}
+
+Mat Math::getSkewSymmetric(const Mat &v_) {
+    const auto * const v = (double *) v_.data;
+    return (Mat_<double>(3,3) << 0, -v[2], v[1],
+            v[2], 0, -v[0],
+            -v[1], v[0], 0);
+}
```

Comment:
`MatCommaInitializer` has poor performance and bloats binary code.

Consider using of `Mat33d` directly.